src/utils/src/Thread.c

```diff
@@ -150,16 +177,71 @@ PUBLIC_API TID defaultGetThreadId()
     return (TID) pthread_self();
 }
 
+PUBLIC_API STATUS defaultCreateThreadWithParams(PTID pThreadId, startRoutine start, SIZE_T stackSize, PVOID args)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    pthread_t threadId;
+    INT32 result;
+    pthread_attr_t* pAttr = NULL;
+
+    CHK(pThreadId != NULL, STATUS_NULL_ARG);
+
+    pthread_attr_t attr;
+    pAttr = &attr;
+    result = pthread_attr_init(pAttr);
+    CHK_ERR(result == 0, STATUS_THREAD_ATTR_INIT_FAILED, "pthread_attr_init failed with %d", result);
+    result = pthread_attr_setstacksize(&attr, stackSize);
+    CHK_ERR(result == 0, STATUS_THREAD_ATTR_SET_STACK_SIZE_FAILED, "pthread_attr_setstacksize failed with %d", result);
+
+    result = pthread_create(&threadId, pAttr, start, args);
+    switch (result) {
+        case 0:
+            // Successful case
+            break;
+        case EAGAIN:
+            CHK(FALSE, STATUS_THREAD_NOT_ENOUGH_RESOURCES);
+        case EINVAL:
+            CHK(FALSE, STATUS_THREAD_INVALID_ARG);
+        case EPERM:
+            CHK(FALSE, STATUS_THREAD_PERMISSIONS);
+        default:
+            // Generic error
+            CHK(FALSE, STATUS_CREATE_THREAD_FAILED);
+    }
+
+    *pThreadId = (TID) threadId;
+
+CleanUp:
+
+    if (pAttr != NULL) {
+        result = pthread_attr_destroy(pAttr);
+        if (result != 0) {
+            DLOGW("pthread_attr_destroy failed with %u", result);
+        }
+    }
+
+    return retStatus;
+}
+
 PUBLIC_API STATUS defaultCreateThread(PTID pThreadId, startRoutine start, PVOID args)
 {
     STATUS retStatus = STATUS_SUCCESS;
     pthread_t threadId;
     INT32 result;
     pthread_attr_t* pAttr = NULL;
+    SIZE_T stackSize = 0;
 
     CHK(pThreadId != NULL, STATUS_NULL_ARG);
 
-#ifdef CONSTRAINED_DEVICE
+#if defined(KVS_DEFAULT_STACK_SIZE)
+    pthread_attr_t attr;
+    pAttr = &attr;
+    result = pthread_attr_init(pAttr);
+    CHK_ERR(result == 0, STATUS_THREAD_ATTR_INIT_FAILED, "pthread_attr_init failed with %d", result);
+    stackSize = KVS_DEFAULT_STACK_SIZE;
+    result = pthread_attr_setstacksize(&attr, stackSize);
```

Comment:
I think we should have a min check here and proactively set the stack size to min if the value is below 16KB. I see we have a check in our WebRTC samples, but instead of returning a fail status, might as well set to a default minimum. What do you think? 

If we do not want to default, it might be useful to have specific status codes for specific result value returned by `pthread_attr_setstacksize`.

Reply:
done